src/account_manager.rs

```diff
@@ -95,11 +119,28 @@ impl AccountManager {
     pub fn start_background_sync(&mut self) {
         if !self.started_monitoring {
             let monitoring_disabled = self.start_monitoring().is_err();
-            self.start_polling(monitoring_disabled);
+            self.polling_handle = Some(self.start_polling(monitoring_disabled));
             self.started_monitoring = true;
         }
     }
 
+    /// Stops the background polling and MQTT monitoring.
+    pub fn stop_background_sync(&mut self) -> crate::Result<()> {
+        {
+            let accounts = self.accounts.read().unwrap();
+            for account in accounts.values() {
+                let _ = crate::monitor::unsubscribe(account.clone());
+            }
+        }
+
+        if let Some(handle) = self.polling_handle.take() {
+            self.stop_polling.store(true, Ordering::Relaxed);
```

Comment:
The polling thread could instead become an async task. Making this change would be better since it might then `select!` over a oneshot channel so that it shuts down immediately instead of when the next polling period comes around. This would both reduce CPU usage (no need for frequent wakeups) and reduce shutdown time (no need to wait for the next wakeup).